src/init.ts

```diff
@@ -18,17 +19,52 @@ const loadDevServerBlock = async (block: Block) => {
   const importContent = imports[importPath];
   const content = await importContent();
 
-  console.log({ content });
-
   let component;
 
   return (props) => {
+    const fullProps = {
+      ...props,
+      BlockComponent: getBlockComponentWithParentContext(props.context),
+    };
     if (component) {
-      component.$set(props);
+      component.$set(fullProps);
     } else {
-      component = new content.default({ target: root, props });
+      component = new content.default({ target: root, props: fullProps });
     }
   };
 };
 
 init(loadDevServerBlock);
+
+const getBlockComponentWithParentContext = (
+  parentContext?: FileContext | FolderContext
+) => {
+  class BlockComponentWithParentContext extends BlockComponent {
+    constructor(options) {
+      let context = {
+        ...(parentContext || {}),
+        ...(options.parentContext || {}),
+      };
+
+      if (parentContext) {
+        // clear sha if viewing content from another repo
+        const parentRepo = [parentContext.owner, parentContext.repo].join("/");
+        const childRepo = [context.owner, context.repo].join("/");
+        const isSameRepo = parentRepo === childRepo;
+        if (!isSameRepo) {
+          context.sha = options.context?.sha || "HEAD";
+        }
+      }
+
+      const fullOptions = {
+        ...options,
+        props: {
+          ...options.props,
+          context,
+        },
+      };
+      super(fullOptions);
+    }
+  }
```

Comment:
we might need to update the props on `$set` as well, I'll take another look once we have this template updating properly on context change